### DISCUSSION TITLE
python-socketio - - - error-handling

### DISCUSSION THREAD

**Discussion by 4n1qz5skwv** [miguelgrinberg/python-socketio#1031] [Q&A] [answered]
Can't find this ([flask-socketio error-handling](https://flask-socketio.readthedocs.io/en/latest/getting_started.html#error-handling)) equivalent section for **python-socketio**. Can someone please link to it. Or does it not exist. if that's the case how do you handle exception? 

I want to emit an event on error from any handler. Please Help!

Thank you.

**Comment by miguelgrinberg** [comment, marked as answer]
This package does not have configurable error handlers. If an error occurs, the stack trace will be logged and that's it. If you need to have custom logic, then you can use a standard Python solution based on try/except to catch errors in your handlers. If you need this in several handlers, you can implement a decorator that does this.

**Reply by 4n1qz5skwv** [reply]
Thanks. Yes. Currently using a [decorator](https://loguru.readthedocs.io/en/stable/api/logger.html#loguru._logger.Logger.catch). But just curious, why skipped it for python-socketio, when it already in flask-socketio? is it bad practice? against design pattern?

**Reply by miguelgrinberg** [reply]
With HTTP an error handler is necessary because you must return a response, even if an error occurs. With Socket.IO this is not the case, so there is no requirement from the framework point of view to handle the error.

In Flask-SocketIO I added a handler to be consistent with Flask error handling practices. This package has no connection to Flask, so I did not see a good reason to do it.

**Reply by 4n1qz5skwv** [reply]
But at least [Server](https://python-socketio.readthedocs.io/en/latest/server.html) needs an Error handler to must return a response. 

For example [socketio.Client.call](https://python-socketio.readthedocs.io/en/latest/api.html#socketio.Client.call) always needs a response, otherwise it will just wait for all the timeout. even when server crashed a long time ago.

@miguelgrinberg

**Reply by miguelgrinberg** [reply]
Responses are always optional, a call timing out is a perfectly acceptable outcome. The point that I was trying to make that maybe you missed is that with Flask, there is a range of default responses that the framework can issue when there is an error. With Socket.IO there is absolutely nothing that can be done by default. In every case an error needs to be handled by the application. So having an error handler is less useful, since it can only work when the application provides the handling code.
